### PR TITLE
Make magithub-comment-reply quote the region if it is active

### LINKED ISSUE #244: When region is active, magithub-comment-reply should only quote region

### PR DESCRIPTION
**Pull request by iravid**
Resolves #244

### CODE REVIEW

**Comment by iravid** [comment]
I just started using Magithub and it's awesome, so thought I'd pick something off the issues list and solve it :-)

**Comment by vermiculus** [comment]
Can't wait to try it out, thanks!!

**Comment by iravid** [comment]
Sure :-) I had to context switch so I couldn’t try it out myself yet, but hopefully tonight :)

**Comment by iravid** [comment]
Whoops, sorry, I misread the code and this isn't doing the right thing yet.

**Comment by iravid** [comment]
Ok, should be good now. I'm not sure it's the cleanest way of doing this as https://stackoverflow.com/a/605931/2535374 says we should be using the `(interactive "r")` form to integrate the start/end points into the function arguments, but it works :-)

File: magithub-comment.el [hunk 1]
```diff
@@ -221,7 +221,10 @@ from COMMENT."
      (or issue (magithub-request (ghubp-follow-get .issue_url)))
      discard-draft
      (with-temp-buffer
-       (insert (string-trim (magithub-wash-gfm .body)))
+       (insert (string-trim (magithub-wash-gfm
+                             (if (use-region-p)
+                                 (buffer-substring (region-beginning) (region-end))
+                                 (.body)))))
```
Comment (vermiculus) [inline magithub-comment.el]: Doesn't seem like `.body` should be in a list, here :)

Comment (vermiculus) [reply magithub-comment.el]: That should resolve [this build error](https://travis-ci.org/vermiculus/magithub/jobs/336091844#L890), too.

Comment (iravid) [reply magithub-comment.el]: Right! Thanks :-) Fixed

**Comment by vermiculus** [comment]
It's not working for me ☹ `use-region-p` is returning nil because `region-active-p` is nil.

Unless I'm doing something incorrectly:
1. Place point at start of region.
2. `C-SPC` to start the selection (i.e., place mark).
3. Move point to the end of my region.
4. Use `r` to call `magithub-comment-reply`.

**Comment by iravid** [comment]
> It's not working for me ☹ `use-region-p` is returning nil because `region-active-p` is nil.
>
> Unless I'm doing something incorrectly:
> 1. Place point at start of region.
> 2. `C-SPC` to start the selection (i.e., place mark).
> 3. Move point to the end of my region.
> 4. Use `r` to call `magithub-comment-reply`.

Hmm, strange. It's not working here too, but `use-region-p` and `region-active-p` are evaluating to `t` when I test them with the region active.

However, I'm using Spacemacs and activating the region using `v` (e.g. Visual Line in evil). Maybe that has something to do with it?

I do remember getting it working at some point ;) Let me dig into this some more.

**Comment by iravid** [comment]
Ahh, it's probably because I moved the call to `use-region-p` into the `with-temp-buffer` form.

**Comment by iravid** [comment]
Ok, give this another try :-)

**Comment by iravid** [comment]
Hey @vermiculus, had a chance to give this another try?

**Comment by vermiculus** [comment]
Sorry it's taken me a little while; work's been a little crazy 😄 I'm looking now.

**Comment by vermiculus** [comment]
Things are looking good!  Posting a [link](http://github.com) to test that.

**Comment by vermiculus** [comment]
Thanks!

**Pull request merged**